urbanairship-react-native/js/UrbanAirship.ts

```diff
@@ -371,47 +384,51 @@ export class UrbanAirship {
   }
 
   /**
-   * Global data collection flag. Enabled by default, unless `dataCollectionOptInEnabled`
-   * is set to `YES` in AirshipConfig.plist on iOS, and `true` in airshipconfig.properties on Android.
-   * When disabled, the device will stop collecting and sending data for named user, events,
-   * tags, attributes, associated identifiers, and location from the device.
-   *
-   * Push notifications will continue to work only if `UrbanAirship.setPushTokenRegistrationEnabled`
-   * has been explicitly set to `true`, otherwise it will default to the current state of `isDataCollectionEnabled`.
+   * Sets the SDK features that will be enabled. The rest of the features will be disabled.
+   * 
+   * If all features are disabled the SDK will not make any network requests or collect data.
    *
-   * @note To disable by default, set the `dataCollectionOptInEnabled` flag to `YES` in AirshipConfig.plist on iOS, and `true` in airshipconfig.properties on Android.
-   * @param enabled true to enable data collection, false to disable.
+   * @note All features are enabled by default.
+   * @param feature An array of `Features` to enable.
+   */
+  static setEnabledFeatures(features: Feature[]) {
+    UrbanAirshipModule.setEnabledFeatures(features);
+  }
+
+  /**
+   * Gets a flag number representing the enabled features.
+   * 
+   * @return A promise that returns the enabled features as a flag `number`.
    */
-  static setDataCollectionEnabled(enabled: boolean) {
-    UrbanAirshipModule.setDataCollectionEnabled(enabled);
+  static getEnabledFeatures(): Promise<number> {
```

Comment:
should getEnabledFeatures return a `Promise<Feature[]>`?

Reply:
Actually I was wondering that too. But since the native SDKs was returning int values I let it like this.
But I think you're right, an array of Feature should be better for the customers.
I'll make the change then.